models/Document/Editable/Video.php

```diff
@@ -36,13 +48,13 @@ class Video extends Model\Document\Editable implements IdRewriterInterface
     protected $id;
 
     /**
-     * one of asset, youtube, vimeo, dailymotion
+     * one of self::ALLOWED_TYPES
      *
      * @internal
      *
      * @var string|null
      */
-    protected $type = 'asset';
+    protected $type;
```

Comment:
If i call `getVideoType()` before `getData()` this will now return `null` as oppose to returning value 'asset' before.. right? 

we should consider setting `$this->type` from `allowedTypes` when `null` in `getVideoType()`.

Reply:
Good catch 👍 

You mean like so? 
```
public function getVideoType()
{
    if (empty($this->type) === true) {
        $this->type = $this->allowedTypes[0];
    }

    return $this->type;
}
```

Reply:
exactly!!

Reply:
Resolved with https://github.com/pimcore/pimcore/pull/12445/commits/11d1ab3bca09acdb87e9a587d03fafdf832c4e0d